### PR TITLE
Clight[d]: Update to 4.2 and 5.1

### PR DESCRIPTION
**Pull request by adriangoransson**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by adriangoransson** [comment]
No idea how to get this running on ARM cross builds. Any help is appreciated! :)

File: srcpkgs/Clight/template [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
 short_desc="Use your webcam as a light sensor to adjust screen backlight"
 maintainer="Dominic Monroe <monroef4@googlemail.com>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/FedeDP/Clight"
 distfiles="https://github.com/FedeDP/Clight/archive/${version}.tar.gz"
-checksum=84156f3dfd9ab863be25d3efc8b5d6d6043ba4a0b44259c2a1cee1367b3c97aa
+checksum=fc3c4952c755bad5f4274f69f5cc17142abb51d88d7c615873e3058edc1c1137
+
+do_install() {
+	vbin build/clight
```
Comment (ericonr) [inline srcpkgs/Clight/template:18]: Why are you overriding this?

File: srcpkgs/Clightd/template [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 pkgname=Clightd
-version=4.2
+version=5.1
 revision=1
 build_style=cmake
 cmake_builddir=build
 configure_args="-DENABLE_DDC=1 -DENABLE_GAMMA=1 -DENABLE_DPMS=1 -DENABLE_SCREEN=1"
 hostmakedepends="pkg-config"
-makedepends="eudev-libudev-devel libmodule-devel polkit-devel elogind-devel libXrandr-devel libXext-devel libX11-devel ddcutil"
+makedepends="eudev-libudev-devel libmodule-devel polkit-devel elogind-devel libXrandr-devel libXext-devel libX11-devel ddcutil libjpeg-turbo-devel wayland-devel libdrm-devel libusb-devel"
```
Comment (ericonr) [inline srcpkgs/Clightd/template:9]: Please break this line at around 80 columns.

File: srcpkgs/Clightd/template [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
 maintainer="Dominic Monroe <monroef4@googlemail.com>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/FedeDP/Clightd"
 distfiles="https://github.com/FedeDP/Clightd/archive/${version}.tar.gz"
-checksum=39301b0cad2da9adb4b662b64563f35ad5b3724631d81c1db55c311bc10dec67
+checksum=ce5002ed7790fce64e323bd483d8e5dde70a7fa9bbebc8311f514e0d5a64ab10
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" wayland-devel"
+fi
```
Comment (ericonr) [inline srcpkgs/Clightd/template:20]: You can add this unconditionally to hostmakedeps, if you want. It's likely necessary due to wayland-scanner.

**Review by ericonr — COMMENTED** [review]
Remember to squash together the commits for each package :)

**Comment by adriangoransson** [comment]
Squashed the latest commit and the PR was closed it seems.

File: srcpkgs/Clight/template (hunk 1 above)
Comment (adriangoransson) [reply srcpkgs/Clight/template]: Ah, mistake from the very beginning. I tried to add shell completions but suddenly the binary disappeared. It wasn't until later that I added those to post_install. :)

**Pull request merged**